install/install.ps1

```diff
@@ -1,94 +1,203 @@
+    $ProcessCheck = Get-Process -Name azureauth -ErrorAction SilentlyContinue -ErrorVariable ProcessError
+    if ($null -ne $ProcessCheck) {
+        Write-Verbose "Stopping any currently running azureauth instances"
+        taskkill /f /im azureauth.exe 2>&1 | Out-Null
+
+        # After killing the process it is still possible for there there to be locks on the files it was using (including
+        # its own DLLs). The OS may take an indeterminate amount of time to clean those up, but so far we've observed 1
+        # second to be enough.
+        Start-Sleep -Seconds 1
+    }
+
+    if (Test-Path -Path $targetDirectory) {
+        Write-Verbose "Removing pre-existing extracted directory at ${targetDirectory}"
+        Remove-Item -Force -Recurse $targetDirectory
+    }
+
+    Write-Verbose "Extracting ${zipFile} to ${targetDirectory}"
+    Add-Type -AssemblyName System.IO.Compression.FileSystem
+    [System.IO.Compression.ZipFile]::ExtractToDirectory($zipFile, $targetDirectory)
+
+    Write-Verbose "Removing ${zipFile}"
+    Remove-Item -Force $zipFile
+
+    if ($NoUpdatePath) {
+        Write-Verbose "Not updating `$env:PATH"
+    }
+    else{
+        $registryPath = 'Registry::HKEY_CURRENT_USER\Environment'
+        $currentPath = (Get-ItemProperty -Path $registryPath -Name PATH -ErrorAction SilentlyContinue).Path
+        $currentAzureauth = (get-command azureauth -ErrorAction SilentlyContinue).Source
+        $currentAzureauthParent = if($null -ne $currentAzureauth) {
+            (get-item $currentAzureauth).Directory.Parent.FullName }
+        
+        $newPath = "";
+
+        if (($null -ne $currentPath) -And ($currentPath.Contains($azureauthDirectory) -Or $currentPath.Contains($currentAzureauthParent))) {
+            $paths = $currentPath.Split(";")
+            $pathArr = @()
+            ForEach($path in $paths){
+                if(!(($path.Equals("")) -Or ($path.Contains($azureauthDirectory)) -Or (($null -ne $currentAzureauthParent) -And $currentAzureauthParent.Contains($path)))){
+                    $pathArr += "${path}"
+                }
+                else {
+                    Write-Verbose "Removing '${path}' from `$env:PATH"
+                }
+            }
+            $pathArr += "${targetDirectory}"
+            $newPath = $pathArr -join ";"
+        }
+        else {
+            Write-Verbose "Appending '${targetDirectory}' to `$env:PATH"
+            if ($null -eq $currentPath) {
+                $newPath = "${targetDirectory}"
+            } else {
+                $newPath = "${currentPath};${targetDirectory}"
+            }
+        }
+
+        setx PATH $newPath > $null
+    }
+
+    Write-Output "Installed azureauth $version!"
+
 }
 
-Write-Output "Installed azureauth $version!"
+switch ($version) {
+     { $_ -in "v0.1.0","v0.2.0","v0.3.0","0.3.1" } {
+        Install-Pre-0-4-0
+    }
+    default {
+        Install-Post-0-4-0
+    }
+}
```

Comment:
We don't need to really consider the installation script version. In the pull request #96 , installation has been detoured to the corresponded tag. There for we only need to consider the install script for current version.

Reply:
@kyle-rader @reillysiemens , I think this part is worth to discuss.

Reply:
@goagain: In #96 we changed our _documentation_, but we can't change any existing user installations that may have been done _before_ we changed that. We **must** consider the version in the installation script to avoid breaking the user's environment if they have a different version installed globally. This is not ideal and I would like for us to do this only for some transition period (maybe 6 months?).